### DISCUSSION TITLE
Error in the build process, navigation with sidebar

### DISCUSSION THREAD

**Discussion by le1erdimanche** [datenstrom/community#252] [Report a bug] [answered]
In the build process to a static website occurs this error:

Fatal error: Call to a member function getHtml() on null in /system/themes/snippets/sidebar.php on line 8

The line 8 of the relevant snippet:

<p><?php echo $page->getHtml("titleNavigation") ?></p>

Versions:
Yellow: latest
PHP 5.6.30
OS: Sierra 10.12.5

**Comment by markseuffert** [comment, marked as answer]
Thanks for the bug report.

It looks like `Navigation: navigation-sidebar` didn't handle non-existing pages and the build process was terminated. Should be working now. Can you test with latest sidebar?

**Comment by le1erdimanche** [comment]
Thank you for the fix:

Yellow build: 113 files, 0 errors, status 200

But the resulting static site isn't usable. Seems like I have to check my setup…

Anyway, Yellow is a nice machine. Thanks a lot for building it!

**Comment by markseuffert** [comment]
Have a look at `StaticUrl` in your `system/config/config.ini` file. 

Hope this helps. 😄

**Comment by le1erdimanche** [comment]
That was the confusion: 

The static site needs a running server for its proper functioning.

Fine now.
